control-plane/connect-inject/controllers/pod/pod_controller.go

```diff
@@ -336,9 +336,11 @@ func (r *Controller) writeWorkload(ctx context.Context, pod corev1.Pod) error {
 	}
 	data := inject.ToProtoAny(workload)
 
+	resourceID := getWorkloadID(pod.GetName(), r.getConsulNamespace(pod.Namespace), r.getPartition())
+	r.Log.Info("registering workload with Consul", getLogFieldsForResource(resourceID)...)
```

Comment:
This adds helpful logging analogous to what we do elsewhere

https://github.com/hashicorp/consul-k8s/blob/2b0ef6ae5a06e7ec274d77f694a39567177140d7/control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller.go#L112

https://github.com/hashicorp/consul-k8s/blob/2b0ef6ae5a06e7ec274d77f694a39567177140d7/control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go#L336